presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java

```diff
@@ -30,6 +30,10 @@
 
     private Block block;
 
+    /**
+     * Creates a {@link LazyBlock} instance. Lazy block loader
+     * should load block partially if possible.
+     */
```

Comment:
I'd put the documentation about the loader on the `LazyBlockLoader` interface

Reply:
done. Does `LazyBlockLoader` still need to accept `T`?

Reply:
I can't remember why I designed the API this way, and I find it really confusing.  I feel like we should be able to replace `LazyBlockLoader` with `Suppler<Block>`